cpp/src/strings/slice.cu

```diff
@@ -59,7 +59,7 @@ struct substring_from_fn {
 
     auto const stop = stops[idx];
     auto const end  = (((stop < 0) || (stop > length)) ? length : stop);
-    return d_str.substr(start, end - start);
+    return start < end ? d_str.substr(start, end - start) : string_view{};
```

Comment:
Should this be moved to the beginning of the function?

Reply:
I don't see how. The `end` value is computed just above this line. This basically an edge case where the values provided are legal within the string but happen to be invalid compared to each other.

Reply:
> The end value is computed just above this line.

Oh I see. I was trying to avoid calling `element<string_view>(idx)` before this but it seems that it is unavoidable.